code.js

```diff
@@ -1,3 +1,54 @@
-function augmentingPath(graph, start, end) {
-    return [];
+function augmentingPath(graph, start, end){
+	fullPath=[]
+	visitedNodes=[]
+	keyValues=Object.keys(graph);
+	source=keyValues[0]
+	for (var i=0; i<Object.keys(graph).length; i++){
+		visitedNodes[keyValues[i]]= Infinity
+	}
+	visitedNodes[source]=0
+	returnedValue=AP(graph, start, end);
+	if (returnedValue==0)
+		return fullPath
+	else if (returnedValue==1)
+		return []
+	else if (returnedValue==2)
+		return []
 }
+
+function AP(graph, currentNode, goalNode){
+	fullPath.push(currentNode)
+	if (goalNode==currentNode){
+		return 0
+	}
+	allVisited=true;
+	visitedKeys=Object.keys(visitedNodes)
+	for (var i=0; i<visitedKeys.length; i++){
+		if (visitedNodes[visitedKeys[i]]===Infinity)
+			allVisited=false;
+	}
+	if (allVisited==true){
+		fullPath=[]
+		return 1
```

Comment:
Hmm, you could just return the path here -- having special return values makes it harder to understand what you're doing.